### ISSUE TITLE
Validateable trait doesn't enforce subclass's constraints

### ISSUE THREAD

**Issue by ZacharyKlein** [grails/grails-core#9595]
Given a base class (not a domain class) that implements the `Validateable` trait, and a subclass that extends the base class and adds it's own constraints, only the constraints in the base class are enforced. 

Interestingly, this is not the case in unit tests - everything works fine there. But in integration tests or actual application code, this issue seems to be pretty consistent. I've created a sample app at https://github.com/ZacharyKlein/validateable-subclass that demonstrates the problem in an integration test (`EmployerIntegrationSpec`) and a controller (`EmployerSpec`). A unit test (`EmployerSpec`), which passes without showing the problem, is also included.

**Comment by ZacharyKlein** [comment]
Workaround: if I implement the `Validateable` trait on the subclass, then constraints from both parent and subclass are applied correctly, in both integration tests and application code.

**Comment by jeffbrown** [comment]
I will look into it.  Probably related to https://issues.apache.org/jira/browse/GROOVY-6765.

**Comment by jeffbrown** [comment]
I won't know for sure until I look at it more closely but this might just be a matter of documenting that subclasses need to be marked with the annotation.

**Comment by ZacharyKlein** [comment]
@jeffbrown Does the `@Validateable` as an annotation exist in Grails 3? I can only find it in the 2.x docs; using it in my 3.0.12 app gives me a `not an annotation` error. The 3.x docs have `implements Validateable`  (which is what I was using in my workaround).

**Comment by jeffbrown** [comment]
> Does @Validateable as an annotation exist in Grails 3?

No.  Ignore my earlier comment.  I was confused myself.  Sorry for the noise.  I even wrote the trait based validation stuff myself and still somehow made the mistake.  ;)

**Comment by stijnvandrunen** [comment]
I've been experiencing the same problem. All my command object inherit from AbstractCommand which implements Validatable as trait. The constraints map of the AbstractCommand is enforced, however the constraints maps of the sub classes that are actually instantiated are completely ignored. Would be nice if getConstraintsMap() also finds the constraints on the instantiated sub classes, as this is how you'd expect it to work :-)

**Comment by ilopmar** [comment]
@ZacharyKlein's integration test now pass in Grails 3.3.8
